awscli/customizations/cloudformation/yamlhelper.py

```diff
@@ -53,12 +54,22 @@ def intrinsics_multi_constructor(loader, tag_prefix, node):
     return {cfntag: value}
 
 
+def str_representer(dumper, node):
+
+    if re.match(r'[-+]?[0-9][0-9_]*', node):
```

Comment:
@DiegoPomares Can we quote all strings by default? Many people are running into this bug. #2556 gives a workaround, but I want to fix the original problem as well for YAML